src/program.cpp

```diff
@@ -143,19 +143,28 @@ void program::compile(const target& t, compile_options options)
 
     options.trace(*this);
     options.trace();
-    auto&& passes = t.get_passes(this->impl->ctx, options);
 
-    auto* modl = get_main_module();
-    assert(modl->validate() == modl->end());
-    run_passes(*modl, passes, options.trace);
-    auto invalid = this->validate();
-    if(invalid != modl->end())
+    auto mods = this->get_modules();
+    std::reverse(mods.begin(), mods.end());
+    auto pass_options = options;
+
+    for(const auto& mod : mods)
     {
-        auto index = std::distance(modl->begin(), invalid);
-        MIGRAPHX_THROW("Invalid module " + modl->name() + " from compilation at instruction " +
-                       std::to_string(index));
+        // submodules do not use offload_copy
+        pass_options.offload_copy = ((mod->name() == "main") ? options.offload_copy : false);
```

Comment:
We shouldn't modify the compile options here.

Reply:
Instead, lowering should check if the module name is `main` to do offload copying.

Reply:
changed accordlingly